zerver/models.py

```diff
@@ -3025,6 +3025,7 @@ class AbstractMessage(models.Model):
     # A JSON-encoded list of objects describing any past edits to this
     # message, oldest first.
     edit_history = models.TextField(null=True)
+    edit_history_entries = models.JSONField(default=list)
```

Comment:
This probably needs to be created with `null=True`, otherwise messages cannot be accessed while this first migration is running, right?

Reply:
From PostgreSQL 11, adding a column with a constant default for a huge table should be [pretty fast](https://www.postgresql.org/docs/current/ddl-alter.html#DDL-ALTER-ADDING-A-COLUMN), so the table won't be locked down. Using the list default should be fine, unless there was another reason that you had in mind?

Reply:
Hmm, but does Django actually use that feature? I seem to recall that the Django ORM generally doesn't set a `DEFAULT` in postgres, instead doing that sort of thing entirely in Python -- which is necessary to have a default that's a function, as you have here.

Reply:
> zerver.ArchivedMessage.edit_history_entries: (fields.E010) JSONField default should be a callable instead of an instance so that it's not shared between all field instances.
        HINT: Use a callable instead, e.g., use `dict` instead of `{}`.

Django gives a warning when the default is not a callable, and it only calls the callable when constructing the migration query:
```sql
$ ./manage.py sqlmigrate zerver 0478
BEGIN;
--
-- Add field edit_history_entries to archivedmessage
--
ALTER TABLE "zerver_archivedmessage" ADD COLUMN "edit_history_entries" jsonb DEFAULT '[]' NOT NULL;
ALTER TABLE "zerver_archivedmessage" ALTER COLUMN "edit_history_entries" DROP DEFAULT;
--
-- Add field edit_history_entries to message
--
ALTER TABLE "zerver_message" ADD COLUMN "edit_history_entries" jsonb DEFAULT '[]' NOT NULL;
ALTER TABLE "zerver_message" ALTER COLUMN "edit_history_entries" DROP DEFAULT;
COMMIT;
```
I notice that they also drop the default, which turns out to be a [design descision](https://code.djangoproject.com/ticket/28000) to utilize the database level default during migration.

The ORM indeed does not use the database level default to run the Django application, which is consistent with your impression, but that only happens after the migration.